src/python/pants/backend/go/util_rules/build_pkg.py

```diff
@@ -226,6 +238,36 @@ class GoCompileActionIdResult:
     action_id: str
 
 
+# TODO: Merge this rule helper and the AssemblyPostCompilationRequest.
```

Comment:
Should this be done before landing? Or at least create a ticket so we don't forget this?

Reply:
I'm fine leaving it as-is for now. Probably merits a separate "link object files" rule, but that can be done in an easily reviewed separate PR. https://github.com/pantsbuild/pants/issues/16831